scripts/doc_checker.py

```diff
@@ -0,0 +1,546 @@
+)
+
+MODIN_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.insert(0, MODIN_PATH)
+
+# error codes that pandas test in CI
+# https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks
+NUMPYDOC_BASE_ERROR_CODES = {
+    *("GL01", "GL02", "GL03", "GL05", "GL06", "GL07", "GL08", "GL09", "GL10"),
+    *("SS02", "SS03", "SS04", "SS05", "PR01", "PR02", "PR03", "PR04", "PR05"),
+    *("PR08", "PR09", "PR10", "RT01", "RT04", "RT05", "SA02", "SA03"),
+}
+
+MODIN_ERROR_CODES = {
+    "MD01": "'{parameter}' description should be '[type], default: [value]', found: '{found}'",
+    "MD02": "Spelling error in line: {line}, found: '{word}', reference: '{reference}'",
+}
+
+
+def get_optional_args(doc: Docstring) -> dict:
+    """
+    Get optional parameters for the object for which the docstring is checked.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    dict
+        Dict with default argument names and its values.
+    """
+    obj = doc.obj
+    if not callable(obj) or inspect.isclass(obj):
+        return {}
+    signature = inspect.signature(obj)
+    return {
+        k: v.default
+        for k, v in signature.parameters.items()
+        if v.default is not inspect.Parameter.empty
+    }
+
+
+def check_optional_args(doc: Docstring) -> list:
+    """
+    Check type description of optional arguments.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+
+    Returns
+    -------
+    list
+        List of tuples with Modin error code and its description.
+    """
+    if not doc.doc_parameters:
+        return []
+    optional_args = get_optional_args(doc)
+    if not optional_args:
+        return []
+
+    errors = []
+    for parameter in optional_args:
+        # case when not all parameters are listed in "Parameters" section;
+        # it's handled by numpydoc itself
+        if parameter not in doc.doc_parameters:
+            continue
+        type_line = doc.doc_parameters[parameter][0]
+        has_default = "default: " in type_line
+        has_optional = "optional" in type_line
+        if not (has_default ^ has_optional):
+            errors.append(
+                (
+                    "MD01",
+                    MODIN_ERROR_CODES["MD01"].format(
+                        parameter=parameter,
+                        found=type_line,
+                    ),
+                )
+            )
+    return errors
+
+
+def check_spelling_words(doc: Docstring) -> list:
+    """
+    Check spelling of chosen words: "Modin", "pandas", "NumPy" in doc.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    list
+        List of tuples with Modin error code and its description.
+
+    Notes
+    -----
+    "modin" word is treated as the constant.
```

Comment:
Please, explain this phrase. I do not fully understand it. Also, why `modin`, but not `Modin`?

Reply:
Any special words enclosed in apostrophes(") are treated as python string constants and are not checked for spelling.

Reply:
Okay, I see.